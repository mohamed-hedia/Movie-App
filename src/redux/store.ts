import { configureStore } from "@reduxjs/toolkit";
import homeSlice from "./home/homeSlice";
import moviesSlice from "./movies/moviesSlice";
import tvSlice from "./tv/tvSlice";
import detailsSlice from "./details/detailsSlice";
import searchSlice from "./search/searchSlice";
import userSlice from "./user/userSlice";
import bookmarkSlice from "./bookmarked/bookmarkSlice";

export const store = configureStore({
  reducer: {
    user: userSlice,
    home: homeSlice,
    movies: moviesSlice,
    tv: tvSlice,
    details: detailsSlice,
    search: searchSlice,
    bookmark: bookmarkSlice,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
