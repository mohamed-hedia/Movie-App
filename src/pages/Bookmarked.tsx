import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";

import GridLayout from "@/components/layout/GridLayout";
import PageLayout from "@/components/layout/PageLayout";
import Heading from "@/components/ui/Heading";
import ItemCard from "@/components/ui/ItemCard";
import { fetchBookmark } from "@/redux/bookmarked/bookmarkSlice";
import { AppDispatch, RootState } from "@/redux/store";

const Bookmarked = () => {
  const { loading, bookmarks } = useSelector(
    (state: RootState) => state.bookmark
  );
  const dispatch = useDispatch<AppDispatch>();

  const movies = bookmarks?.filter((item) => item.media_type === "movie");
  const tvShows = bookmarks?.filter((item) => item.media_type === "tv");

  useEffect(() => {
    dispatch(fetchBookmark());
  }, [dispatch]);

  return (
    <PageLayout loading={loading}>
      {!loading && (
        <>
          <section>
            <Heading as="h2" className="mt-6">
              Bookmarked Movies
            </Heading>
            {movies && movies?.length !== 0 ? (
              <GridLayout>
                {movies?.map((movie) => (
                  <ItemCard
                    key={movie.id}
                    id={movie.id}
                    imgSrc={movie.backdrop_path}
                    releaseDate={movie.release_date?.substring(0, 4)}
                    media_type="movie"
                    ratings={movie.adult ? "18+" : "PG"}
                    title={movie.title}
                  />
                ))}
              </GridLayout>
            ) : (
              <div className="">No Bookmarked Movies...</div>
            )}
          </section>
          <section>
            <Heading as="h2" className="mt-6">
              Bookmarked TV Shows
            </Heading>
            {!loading && tvShows && tvShows?.length !== 0 ? (
              <GridLayout>
                {tvShows?.map((tvShow) => (
                  <ItemCard
                    key={tvShow.id}
                    id={tvShow.id}
                    imgSrc={tvShow.backdrop_path}
                    releaseDate={tvShow.first_air_date?.substring(0, 4)}
                    media_type="tv"
                    ratings={tvShow.adult ? "18+" : "PG"}
                    title={tvShow.name}
                  />
                ))}
              </GridLayout>
            ) : (
              <div className="">No Bookmarked TV Shows...</div>
            )}
          </section>
        </>
      )}
    </PageLayout>
  );
};
export default Bookmarked;
