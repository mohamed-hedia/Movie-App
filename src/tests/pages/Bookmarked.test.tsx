
import Bookmarked from "@/pages/Bookmarked";
import { store } from "@/redux/store";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";


describe("Bookmarked Page", () => {
  test("renders Bookmarked Page should match the screenshot", () => {
    const bookmarked = render(
      <Provider store={store}>
        <Router>
          <Bookmarked />
        </Router>
      </Provider>
    );

    expect(bookmarked).toMatchSnapshot();
  });
});
