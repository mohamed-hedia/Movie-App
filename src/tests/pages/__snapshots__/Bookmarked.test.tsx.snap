// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bookmarked Page > renders Bookmarked Page should match the screenshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="w-full md:w-[calc(100%-8rem)] min-h-screen py-6 md:ml-32 pl-5 md:px-0"
      >
        <div
          class="h-[calc(100vh-12rem)] md:h-full w-full flex justify-center items-center md:pr-6"
          data-testid="loading-component"
        >
          <div
            class="w-12 h-12 md:w-14 md:h-14 border-[5px] border-t-orange rounded-full border-[#ffffff90] animate-spin"
          />
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="w-full md:w-[calc(100%-8rem)] min-h-screen py-6 md:ml-32 pl-5 md:px-0"
    >
      <div
        class="h-[calc(100vh-12rem)] md:h-full w-full flex justify-center items-center md:pr-6"
        data-testid="loading-component"
      >
        <div
          class="w-12 h-12 md:w-14 md:h-14 border-[5px] border-t-orange rounded-full border-[#ffffff90] animate-spin"
        />
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
